Javascript Implementations

ashcas23 opened the pull request:
Javascript implementations were added in order for the user to be able to interact with the blog page and make it seem more dynamic.

ashleycmr23 committed:
javascript changes made in editor

ashleycmr23 committed:
Javascript Changes Scroll To Top Function

dchukhin commented on blog/static/js/main.js:
```diff
@@ -0,0 +1,33 @@
+function ChangeColorofPostTitles() { //defining a (python) function
```
```suggestion
function ChangeColorofPostTitles() { //defining a (JavaScript) function
```

dchukhin commented on blog/static/js/main.js:
```diff
@@ -0,0 +1,33 @@
+function ChangeColorofPostTitles() { //defining a (python) function
+  let elements_posttitles = document.getElementsByClassName("posttitle");
+
+  for (element of elements_posttitles){
```
Could you add a comment about what this loop is doing?

dchukhin commented on blog/templates/blog/post_list.html:
```diff
@@ -7,8 +7,15 @@
 {% endblock %}
+{% block nextsection %}
+<div class= "row">
+  <button id= "ChangeColorofTitles" class = "btn btn-secondary">Change Blog Titles</button>
```
For clarity, could you rename the id of this button to something with the word 'button'?

dchukhin commented on blog/templates/blog/post_list.html:
```diff
@@ -7,8 +7,15 @@
+{% block nextsection %}
+<div class= "row">
+  <button id= "ChangeColorofTitles" class = "btn btn-secondary">Change Blog Titles</button>
+  <button id= "ChangeColorofMainTitle" class = "btn btn-secondary">Change Main Title</button>
```
For clarity, could you rename the id of this button to something with the word 'button'?

dchukhin reviewed (commented):
Looks good. I left a few comments about potential improvements

ashcas23 committed:
Change comment

Change from Python to JavaScript since it is a JavaScript Function

Co-authored-by: Dmitriy Chukhin <dchukhin@caktusgroup.com>

ashcas23 committed:
Update main.js

ashcas23 committed:
Update post_list.html

ashcas23 committed:
Changed Title To Implement Button In Title

(Added button at the end of Titles to make it more clear to viewer)

dchukhin reviewed (commented):
Good job making the changes. Now that the HTML elements have new ids (for example, ChangeColorTitlesButton), could you update the main.js, so that the JavaScript functions find the appropriate elements (by the new ids)?

ashcas23 committed:
Updated main.js

Updated the names on the main.js file so the Javascript functions find the appropriate elements (new ids).

ashleycmr23 committed:
javascript implementations regarding buttons

ashleycmr23 committed:
javascript merge changes from files

dchukhin commented on blog/static/js/main.js:
```diff
@@ -0,0 +1,39 @@
+ScrollToTopButton.addEventListener("click", ScrollToTopFunction);
+
+function ScrollToBottomFunction() {
+  let ScrollToTopButton = document.getElementById("ScrollToTopButton"); //make sure to change names of titles 
```
Should this be `let ScrollToBottomButton = document.getElementById("ScrollToBottomButton");`?

dchukhin replied:
Ah, I actually remembered why this is the way it is. Could you add a comment explaining it instead?

ashleycmr23 committed:
changed name error in post.list

ashcas23 committed:
Update main.js

dchukhin reviewed (approved):
Looks good!
